### PR TITLE
Add CTX_set_ecdh_auto

### PR DESCRIPTION
**Pull request by TakayukiMatsuo**
For openssl compat layer following funcs are added:
- wolfSSL_THREADID_current
- wolfSSL_THREADID_hash
- wolfSSL_CTX_set_ecdh_auto

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -53643,4 +53643,23 @@ int wolfSSL_set_ephemeral_key(WOLFSSL* ssl, int keyAlgo,
 
 #endif /* WOLFSSL_STATIC_EPHEMERAL */
 
+#if defined(OPENSSL_EXTRA)
+void wolfSSL_THREADID_current(WOLFSSL_CRYPTO_THREADID* id)
```
Comment (cconlon) [inline src/ssl.c:53647]: All wolfSSL API should have at least a one line comment describing what the function does, and in this case, why wolfSSL doesn't need any specific behavior here.

File: src/ssl.c [hunk 2]
```diff
@@ -53643,4 +53643,23 @@ int wolfSSL_set_ephemeral_key(WOLFSSL* ssl, int keyAlgo,
...
+{
+    (void)id;
+    return;
+}
+unsigned long wolfSSL_THREADID_hash(const WOLFSSL_CRYPTO_THREADID* id)
```
Comment (cconlon) [inline src/ssl.c:53652]: Please add comment with function explanation and reason why wolfSSL only returns 0UL here.

File: src/ssl.c [hunk 3]
```diff
@@ -53643,4 +53643,23 @@ int wolfSSL_set_ephemeral_key(WOLFSSL* ssl, int keyAlgo,
...
+{
+    (void)id;
+    return 0UL;
+}
+int wolfSSL_CTX_set_ecdh_auto(WOLFSSL_CTX* ctx, int onoff)
```
Comment (cconlon) [inline src/ssl.c:53657]: Please add function comment on what function does and why wolfSSL doesn't need to take any action in this function, thanks.

**Review by cconlon — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (TakayukiMatsuo) [reply src/ssl.c:53647]: Added comment

File: src/ssl.c (hunk 2 above)
Comment (TakayukiMatsuo) [reply src/ssl.c:53652]: Added comment

File: src/ssl.c (hunk 3 above)
Comment (TakayukiMatsuo) [reply src/ssl.c:53657]: Added comment

**Review by cconlon — APPROVED** [review]

**Comment by cconlon** [comment]
Please fix merge conflicts, thanks!

**Pull request merged**